File: app/src/main/java/com/pcs/ztqtj/view/fragment/FragmentLife.java
package com.pcs.ztqtj.view.fragment;

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.GridView;
import android.widget.ImageView;

import com.pcs.ztqtj.MyApplication;
import com.pcs.ztqtj.R;
import com.pcs.ztqtj.control.adapter.AdapterLifeFragment;
import com.pcs.ztqtj.control.tool.CommUtils;
import com.pcs.ztqtj.control.tool.SpaceItemDecoration;
import com.pcs.ztqtj.view.activity.life.ActivityChannelList;
import com.pcs.ztqtj.view.activity.life.ActivityMeteorologicalScience;
import com.pcs.ztqtj.view.activity.life.expert_interpretation.ActivityExpertList;
import com.pcs.ztqtj.view.activity.life.health.ActivityHealthWeather;
import com.pcs.ztqtj.view.activity.life.travel.ActivityTravelView;
import com.pcs.ztqtj.view.activity.product.media.ActivityMediaList;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * 生活气象
 */
public class FragmentLife extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_live, container, false);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initView();
        initGradView();
    }

    private void initView() {
        ImageView rl = getView().findViewById(R.id.rl);
        rl.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getActivity().startActivity(new Intent(getActivity(), ActivityHealthWeather.class));
            }
        });
    }

    /* 初始化条目 */
    private void initGradView() {
        RecyclerView recycleview = getView().findViewById(R.id.recycleview);
        String[] life_list = getResources().getStringArray(R.array.life_list);
        TypedArray ar = getResources().obtainTypedArray(R.array.life_icon_list);
        TypedArray arGray = getResources().obtainTypedArray(R.array.life_icon_list_gray);
        int[] life_icon_list = new int[ar.length()];
        int[] life_icon_list_gray = new int[arGray.length()];
        for (int i = 0; i < ar.length(); i++) {
            life_icon_list[i] = ar.getResourceId(i, 0);
            life_icon_list_gray[i] = arGray.getResourceId(i, 0);
        }
        ar.recycle();

        List<Map<String, Object>> data = new ArrayList<>();
        for (int i = 0; i < life_list.length; i++) {
            String[] item = life_list[i].split(",");
            Map<String, Object> map = new HashMap<>();
            map.put("t", item[0]);
            if (MyApplication.LIMITINFO.contains(item[1])) {
                map.put("i", life_icon_list[i]);
            } else {
                map.put("i", life_icon_list_gray[i]);
            }
            data.add(map);
        }
        recycleview.addItemDecoration(new SpaceItemDecoration(CommUtils.Dip2Px(getActivity(), 25), CommUtils.Dip2Px(getActivity(), 25), 3));
        recycleview.setAdapter(new AdapterLifeFragment(getActivity(), data));
        recycleview.setLayoutManager(new GridLayoutManager(getActivity(), 3));
    }

    /**
     * java.lang.IllegalStateException: No activity
     * 错误解决方案
     */
    @Override
    public void onDetach() {
        super.onDetach();
        try {
            Field childFragmentManager = Fragment.class.getDeclaredField("mChildFragmentManager");
            childFragmentManager.setAccessible(true);
            childFragmentManager.set(this, null);
        } catch (NoSuchFieldException e) {
            throw new RuntimeException(e);
        } catch (IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    }

}

File: app/src/main/java/com/pcs/ztqtj/view/fragment/FragmentLife.kt
package com.pcs.ztqtj.view.fragment

import android.content.Intent
import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView.OnItemClickListener
import com.pcs.ztqtj.R
import com.pcs.ztqtj.control.adapter.AdapterLifeFragment
import com.pcs.ztqtj.util.ColumnDto
import com.pcs.ztqtj.util.CommonUtil
import com.pcs.ztqtj.view.activity.life.ActivityChannelList
import com.pcs.ztqtj.view.activity.life.ActivityMeteorologicalScience
import com.pcs.ztqtj.view.activity.life.expert_interpretation.ActivityExpertList
import com.pcs.ztqtj.view.activity.life.health.ActivityHealthWeather
import com.pcs.ztqtj.view.activity.life.travel.ActivityTravelView
import com.pcs.ztqtj.view.activity.product.media.ActivityMediaList
import kotlinx.android.synthetic.main.fragment_live.*
import java.util.*

/**
 * 生活气象
 */
class FragmentLife : BaseFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_live, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initGradView()
    }

    private fun initGradView() {
        val itemHeight = (CommonUtil.heightPixels(activity)-CommonUtil.dip2px(activity, 100f).toInt()
                -CommonUtil.statusBarHeight(activity)-CommonUtil.navigationBarHeight(activity)) / 3
        val dto: ColumnDto = arguments!!.getParcelable("data")
        val dataList = ArrayList<ColumnDto>()
        if (dto != null && dto.childList.size > 0) {
            dataList.addAll(dto.childList)
        }
        val mAdapter = AdapterLifeFragment(activity, dataList, itemHeight)
        gridView.adapter = mAdapter
        gridView.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            val data = dataList[position]
            if (!CommonUtil.isCanAccess(data.flag)) {
                return@OnItemClickListener
            }
            val intent = Intent()
            when (data.dataCode) {
                "401" -> {
                    // 旅游气象
                    intent.setClass(activity, ActivityTravelView::class.java)
                    intent.putExtra("title", data.dataName)
                    activity!!.startActivity(intent)
                }
                "402" -> {
                    // 气象影视
                    intent.setClass(activity, ActivityMediaList::class.java)
                    intent.putExtra("title", data.dataName)
                    activity!!.startActivity(intent)
                }
                "403" -> {
                    //天气新闻
                    intent.setClass(activity, ActivityChannelList::class.java)
                    intent.putExtra("title", data.dataName)
                    intent.putExtra("channel_id", "100005")
                    activity!!.startActivity(intent)
                }
                "404" -> {
                    //专家解读
                    intent.setClass(activity, ActivityExpertList::class.java)
                    intent.putExtra("title", data.dataName)
                    activity!!.startActivity(intent)
                }
                "405" -> {
                    //灾害防御
                    intent.setClass(activity, ActivityChannelList::class.java)
                    intent.putExtra("title", data.dataName)
                    intent.putExtra("channel_id", "100007")
                    intent.putExtra("interfaceUrl", "zhfy")
                    activity!!.startActivity(intent)
                }
                "406" -> {
                    // 气象科普
                    intent.setClass(activity, ActivityMeteorologicalScience::class.java)
                    intent.putExtra("title", data.dataName)
                    activity!!.startActivity(intent)
                }
                "407" -> {
                    // 健康气象
                    intent.setClass(activity, ActivityHealthWeather::class.java)
                    intent.putExtra("title", data.dataName)
                    activity!!.startActivity(intent)
                }
            }
        }
    }

}
